meinberlin/apps/kiezradar/views.py

```diff
@@ -0,0 +1,24 @@
+from rest_framework import viewsets
+
+from .models import SearchProfile
+from .serializers import SearchProfileSerializer
+from .signals import add_kiezradars_to_profile
+
+
+class SearchProfileViewSet(viewsets.ModelViewSet):
+    queryset = SearchProfile.objects.all()
+    serializer_class = SearchProfileSerializer
+
+    def perform_create(self, serializer):
+        # Create the SearchProfile instance
+        search_profile = serializer.save()
+
+        # Define topics or retrieve from request data
+        topics = self.request.data.get("topics", ["Default Topic 1", "Default Topic 2"])
+
+        # Send the custom signal with topics
+        add_kiezradars_to_profile.send(
```

Comment:
Why do we need a signal here? can't we just add the topics to the `search_profile` directly?

Reply:
Because of the m2m relation, search profile needs to be saved first in order to add topics, that's why the signal is `post_save`.